src/Service/VendorService/OverDrive/Api/Client.php

```diff
@@ -0,0 +1,316 @@
+    }
+
+    /**
+     * Set OverDrive authentication credentials.
+     *
+     * @see https://developer.overdrive.com/apis/client-auth
+     * @see https://developer.overdrive.com/apis/library-account
+     *
+     * @param string $libraryAccountEndpoint
+     *   The custom endpoint for the account used
+     * @param string $clientId
+     *   Client id
+     * @param string $clientSecret
+     *   Client secret
+     */
+    public function setCredentials(string $libraryAccountEndpoint, string $clientId, string $clientSecret): void
+    {
+        $this->libraryAccountEndpoint = $libraryAccountEndpoint;
+        $this->clientId = $clientId;
+        $this->clientSecret = $clientSecret;
+    }
+
+    /**
+     * Get cover url from OverDrive crossRefId.
+     *
+     * @param string $crossRefId
+     *   The OverDrive 'crossRefId'
+     *
+     * @return string|null
+     *   The cover url or null
+     *
+     * @throws GuzzleException
+     * @throws AuthException
+     * @throws InvalidArgumentException
+     */
+    public function getCoverUrl(string $crossRefId): ?string
+    {
+        $requests = 0;
+
+        // The OverDrive api fails at random but still gives a 200 response.
+        // Only indication of failure is that the 'products' key is missing
+        // from the response body.
+        // If the products key is missing we retry the request 'MAX_RETRIES' times
+        do {
+            $response = $this->httpClient->request('GET', $this->getProductsEndpoint(), [
+                'headers' => $this->getHeaders(),
+                'query' => [
+                    'crossRefId' => $crossRefId,
+                ],
+            ]);
+
+            $content = $response->getBody()->getContents();
+            $json = json_decode($content, false);
+
+            // Check for the product and images keys.
+            $product = isset($json->products) && is_array($json->products) ? array_shift($json->products) : null;
+            $images = $product->images ?? null;
+
+            ++$requests;
+        } while (!$images && $requests < self::MAX_RETRIES);
+
+        return $images ? $images->cover->href : null;
+    }
+
+    /**
+     * Get products from the overdrive api.
+     *
+     * @param int $limit
+     *   The number of products to fetch
+     * @param int $offset
+     *   The offset to fetch from
+     *
+     * @return array
+     *   Array of 'products' serialized as stdClass
+     *
+     * @throws AuthException
+     * @throws GuzzleException
+     * @throws InvalidArgumentException
+     */
+    public function getProducts(int $limit, int $offset): array
+    {
+        $response = $this->httpClient->request('GET', $this->getProductsEndpoint(), [
+            'headers' => $this->getHeaders(),
+            'query' => ['limit' => $limit, 'offset' => $offset],
+        ]);
+
+        $content = $response->getBody()->getContents();
+        $content = json_decode($content);
+
+        return $content->products;
```

Comment:
If the request fails the `json_decode` will return false and accessing property on false will throw an exception.